### ISSUE TITLE
Comand Definition Language

### ISSUE THREAD

**Issue by misos** [amzn/amazon-dsstne#39]
Hello,

we are trying to test DSSTNE on our data and were experimenting with various NN settings.
According to your CDL file, 
https://github.com/amznlabs/amazon-dsstne/blob/master/docs/getting_started/CDL.txt
it should be possible to change optimization algorithm or to set other training parameters. We added the configuration as part of NN json configuration file but it did not work. 

We were getting error "LoadNeuralNetworkJSON: Unknown neural network field: trainingparameters"

Basically we just added block like this to NN json config:
    "TrainingParameters": {
        "epochs": 40, 
        "alpha": 0.01, 
        "optimizer": "Nesterov", 
        "lambda": 0.001
    }

What is the right way to set up training parameters such as optimizer?

Thanks,

Michal

**Comment by rgeorgej** [comment]
In the train executable which is provided i believe that training parameters are not honored.. Currently it has to be configured through the code as mentioned  at https://github.com/amznlabs/amazon-dsstne/blob/master/docs/getting_started/userguide.md#optimization 

Thanks for the feedback and we will push this code out soon

**Comment by misos** [comment]
thanks

**Comment by ekandrotA9** [comment]
Has there be a push of code to address this?  Over a year old, looking get things resolved

**Comment by ekandrotA9** [comment]
It appears that this hasn't been implemented, because main.cpp as well as everything under utils are examples on how to call the DSSTNE engine.  I agree that this looks like something that should be in this repo, to make it easier to use DSSTNE, so I will add the code for this now.

My proposal:  modify main.cpp in utils to parse the format contained within the CDL.  And modify the CDL to match supported commands.  I'll mark the code changes against this issue, #39.  I'll see if I can get this done this weekend, or so.

**Comment by ekandrotA9** [comment]
Reviewed and checked in
